rfd/0092-error-message-guidelines.md

```diff
@@ -0,0 +1,552 @@
+---
+authors: Jeff Anderson <jeff@goteleport.com>
+state: draft
+---
+
+# RFD 0092 - Error Message Guidelines
```

Comment:
I think it would be nice to make some cursory reference to Structured Logging since this does change the information contained within the error message. From an operations PoV, I'm a big fan of it for reasons of being able to easily search/metrickify logs, but I'd be interested in whether we feel that structured logging may be more complicated for a user to understand compared to including the values within the message.